igor-web/src/main/java/com/netflix/spinnaker/igor/health/EchoServiceHealthIndicator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License')
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.health;
+
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.patterns.PolledMeter;
+import com.netflix.spinnaker.igor.build.model.GenericBuild;
+import com.netflix.spinnaker.igor.build.model.GenericProject;
+import com.netflix.spinnaker.igor.history.EchoService;
+import com.netflix.spinnaker.igor.history.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Component
+public class EchoServiceHealthIndicator implements HealthIndicator {
+    private Logger log = LoggerFactory.getLogger(EchoServiceHealthIndicator.class);
+    private final AtomicBoolean upOnce;
+    private final Optional<EchoService> echoService;
+    private final AtomicLong errors;
+
+    static final GenericBuildEvent event = buildGenericEvent();
+
+    @Autowired
+    EchoServiceHealthIndicator(Registry registry,
+                               EchoService echoService) {
```

Comment:
If echo is disabled, are you actually going to get a null `echoService`? I thought it would have failed to launch saying it couldn't find a bean (in other spots we've done the equivalent of `Optional<EchoService> echoService` as the constructor parameter to get around this.